### PR TITLE
Deprecate Project Templates (EOL 2019)

### LINKED ISSUE #917: Deprecate Project Templates (EOL 2019)
I.e. `ProjectTemplateService` and related objects.
See https://support.gooddata.com/hc/en-us/articles/360016126334-April-4-2019

### PR DESCRIPTION
**Pull request by liry**
Fixes #917

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java-model/src/main/java/com/gooddata/sdk/model/projecttemplate/Template.java [hunk 1]
```diff
@@ -19,7 +19,12 @@
 /**
  * Represents one project template.
  * Deserialization only.
+ *
+ * @deprecated The project templates are deprecated and stopped working on May 15, 2019.
+ * See https://support.gooddata.com/hc/en-us/articles/360016126334-April-4-2019
+ * Deprecated since version 3.0.0. Will be removed in one of future versions.
```
Comment (peter-plochan) [inline gooddata-java-model/src/main/java/com/gooddata/sdk/model/projecttemplate/Template.java:25]: Well, we already released the `3.0.0` so we have 2 options:
* It was missing in the `3.0.0` so it's a bugfix
* It's a new feature, so we should say that it's deprecated from `3.1.0`

Comment (liry) [reply gooddata-java-model/src/main/java/com/gooddata/sdk/model/projecttemplate/Template.java:25]: Well, even if I make bugfix out of it, I should change it to 3.0.1 ;)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/164/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-java-fossa-scanning-zuul-el7/4/ : SUCCESS in 15s (non-voting)

**Pull request merged**